<a href="https://colab.research.google.com/github/AkshatkMalik/Python_Programming/blob/main/Destructive-Testing-Estimation-and-CI.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**ASSIGNMENT-2**

**Estimation And Confidence Intervals**



**Task A: Building a 99% Confidence Interval Using the Sample Standard Deviation**

In [1]:
import math

# The data
durability_data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

# Basic counts and stats
n = len(durability_data)
mean_val = sum(durability_data) / n

# Calculateing Standard Deviation
variance = sum((x - mean_val) ** 2 for x in durability_data) / (n - 1)
sample_std = math.sqrt(variance)

print(f"Sample Size (n): {n}")
print(f"Sample Mean: {mean_val:.4f} million characters")
print(f"Sample Standard Deviation (s): {sample_std:.4f}")

# For 99% confidence interval
# the t-critical value is approx 2.9768
t_critical = 2.9768

# Calculating Margin of Error
margin_of_error = t_critical * (sample_std / math.sqrt(n))

# Calculating Lower and Upper Bounds
lower_bound = mean_val - margin_of_error
upper_bound = mean_val + margin_of_error

print(f"\nRESULTS FOR TASK A ")
print(f"Margin of Error: ±{margin_of_error:.4f}")
print(f"99% Confidence Interval: ({lower_bound:.4f}, {upper_bound:.4f})")

Sample Size (n): 15
Sample Mean: 1.2387 million characters
Sample Standard Deviation (s): 0.1932

RESULTS FOR TASK A 
Margin of Error: ±0.1485
99% Confidence Interval: (1.0902, 1.3871)


Sample Mean $(\bar{x}=1.2387)$: On average, our 15 burnt print heads failed after printing 1.24 million characters.

99% Confidence Interval $((1.0902,1.3871))$: Since we only tested a very small batch of 15, we cannot say with 100% confidence what is the average value of all the print heads that could have been manufactured.

Nevertheless, we can say with 99% confidence that the true mean falls within the range of 1.09 to 1.39 million characters.

**Task B: Building a 99% Confidence Interval Using a Known Population Standard Deviation of 0.2 million**

In [2]:
import math

# Given data
durability_data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n = len(durability_data)
mean = sum(durability_data) / n

# Known population std
population_std = 0.2

# For a 99% confidence internal
# the Z-critical value is approxx 2.5758
z_critical = 2.5758

# Calculating Margin of Error
margin_error= z_critical * (population_std / math.sqrt(n))

# Calculating Lower and Upper Bounds
l_bound_b = mean - margin_error
u_bound_b = mean + margin_error

print(f" RESULTS FOR TASK B ")
print(f"Known Population Std Dev (sigma): {population_std}")
print(f"Margin of Error: ±{margin_error:.4f}")
print(f"99% Confidence Interval (Known Sigma): ({l_bound_b:.4f}, {u_bound_b:.4f})")

 RESULTS FOR TASK B 
Known Population Std Dev (sigma): 0.2
Margin of Error: ±0.1330
99% Confidence Interval (Known Sigma): (1.1057, 1.3717)


Approach: Uses a 99% confidence interval constructed using the Normal ($Z$) distribution as opposed to the t-distribution, where the population standard deviation $\sigma = 0.2$ is known.

Formula: $\bar{x} \pm \left( Z^* \times \frac{\sigma}{\sqrt{n}} \right)$, where $Z^* = 2.576$.

Effect: Since you are sure of the population spread as compared to a sample of size 15, the margin of error reduces, and the confidence interval becomes tighter ($\pm 0.1330$ as opposed to $\pm 0.1485$ in Task A).

Task B gives a confidence interval that is a little bit narrower than the interval in Task A ($(1.1057, 1.3717)$ vs $(1.0902, 1.3871)$).

This happens due to the known population standard deviation ($\sigma = 0.2$), which means that we do not need to account for additional error caused by using an estimate of $\sigma$ calculated on the basis of a small sample ($n=15$).